guides/image-sources/provisioning-image-cloudinit.plantuml

```diff
@@ -0,0 +1,55 @@
+@startuml
+
+!include foreman.pstyle
+
+title Image-based provisioning with cloud-init configuration
+
+actor       User
+participant "Provisioned\nInstance" as Host
+participant "Foreman" as Foreman
+participant "Foreman\nProxy" as Proxy
+participant "Infrastructure\nCloud" as Cloud
+participant DNS
+!if ($puppet)
+participant "Puppet\nserver" as Puppet
+!endif
+
+note over Foreman : has an image\nwith cloud-init flag
+
+== Create host in Foreman ==
+
+User -> Foreman : clicks **Create Host**
+User -> Foreman : selects cloud resource,\nthe image and submits
+group Template [cloud-init]
+   Foreman -> Proxy : renders the cloud-init script
+end
+Foreman -> Cloud : creates new instance\nwith cloud-init info
+Cloud -> Foreman : reports IP address
```

Comment:
Technically it can report IPv4 and IPv6. If both are reported, that also means the DNS record will get both `A` and `AAAA` records:
https://github.com/theforeman/foreman/blob/e80d5558767bb713c13264e821122baea577af76/app/models/concerns/orchestration/compute.rb#L69-L72

After this will store the instance details that the compute resource provided:
https://github.com/theforeman/foreman/blob/e80d5558767bb713c13264e821122baea577af76/app/models/concerns/orchestration/compute.rb#L73-L74

Not all compute resources provide IPs and some provide MAC addresses. In that case it will also try to reserve an IP:
https://github.com/theforeman/foreman/blob/e80d5558767bb713c13264e821122baea577af76/app/models/concerns/orchestration/compute.rb#L75-L78

Reply:
> Technically it can report IPv4 and IPv6. 

That is not the level of detail we want here.

> After this will store the instance details that the compute resource provided
>Not all compute resources provide IPs and some provide MAC addresses. In that case it will also try to reserve an IP 

I don't understand what it means for the diagram.

Reply:
> That is not the level of detail we want here.

It's about singular vs plural.

```suggestion
Cloud -> Foreman : reports IP addresses
```

> I don't understand what it means for the diagram.

Probably not for this specific diagram.